nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/MonitorActivity.java

```diff
@@ -105,6 +105,14 @@ public class MonitorActivity extends AbstractProcessor {
             .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
             .defaultValue("Activity restored at time: ${now():format('yyyy/MM/dd HH:mm:ss')} after being inactive for ${inactivityDurationMillis:toNumber():divide(60000)} minutes")
             .build();
+    public static final PropertyDescriptor WAIT_FOR_ACTIVITY = new PropertyDescriptor.Builder()
+            .name("Wait For Activity")
+            .description("When the processor gets started or restarted, if set to true, only send an inactive indicator if there had been activity beforehand. "
+                    + "Otherwise send an inactive indicator even if there had not been activity beforehand.")
+            .required(false)
```

Comment:
Since this property has a default value that preserves the existing behavior, this new property can be marked as required.
```suggestion
            .required(true)
```